WinUIGallery/ControlPages/TitleBarPage.xaml.cs

```diff
@@ -146,9 +150,28 @@ public void UpdateTitleBarColor()
         {
             var window = WindowHelper.GetWindowForElement(this);
             var titleBarElement = UIHelper.FindElementByName(this, "AppTitleBar");
+            var titleBarAppNameElement = UIHelper.FindElementByName(this, "AppTitle");
 
             (titleBarElement as Border).Background = new SolidColorBrush(currentBgColor); // changing titlebar uielement's color
+            if(currentFgColor != Colors.Transparent)
+            {
+                (titleBarAppNameElement as TextBlock).Foreground = new SolidColorBrush(currentFgColor);
+            }
+            else
+            {
+                (titleBarAppNameElement as TextBlock).Foreground = Application.Current.Resources["TextFillColorPrimaryBrush"] as SolidColorBrush;
+            }
             TitleBarHelper.SetCaptionButtonColors(window, currentFgColor);
+            if(currentBgColor == Colors.Transparent)
+            {
+                // If the current background is null, we want to revert to the default titlebar which is achieved using null as color
+                TitleBarHelper.SetBackgroundColor(window, null);
+            }
+            else
+            {
+                TitleBarHelper.SetBackgroundColor(window, currentBgColor);
+            }
+            TitleBarHelper.SetForegroundColor(window, currentFgColor);
```

Comment:
>SetForegroundColor [](http://example.com/codeflow?start=27&length=18)

nit: Capitalize first letter and add period to comment.

Reply:
I'm afraid I'm not sure which first letter to capitalize here 😅